Sources/FoundationEssentials/URL/URITemplate_VariableName.swift

```diff
@@ -0,0 +1,36 @@
+extension URL.Template {
+    /// The name of a variable used for expanding a template.
+    public struct VariableName: Sendable, Hashable {
+        let key: String
+
+        public init(_ key: String) {
+            self.key = key
+        }
+
+        init(_ key: Substring) {
+            self.key = String(key)
+        }
+    }
+}
+
+// MARK: -
+
+extension String {
+    public init(_ key: URL.Template.VariableName) {
```

Comment:
Was this part of the proposal? Adding a public, no-arg init to `String` seems very broad.

Reply:
Oh, I see it now. I'm not sure about it now that I see it in code. I know it is strongly typed, but it can cause confusion in documentation or at call sites. How critical is it to do this conversion in this way at a call site?

Reply:
@parkera would removing this from the implementation require another round of review/block this PR? Or could we maybe mark it internal for now?

Reply:
I think the right way forward is to remove `ExpressibleByStringLiteral`. This is only truly useful in small demo code. In “real” code, you’d usually have a fixed set of variable names, and would probably do something like:
```swift
extension URL.Template.VariableName {
    static var query: URL.Template.VariableName { URL.Template.VariableName("query") }
    static var number: URL.Template.VariableName { URL.Template.VariableName("number") }
}
```
and then use that when constructing the dictionary.

Reply:
Agreed, or even

```
static var query: URL.Template.VariableName { .init("query") }
```